src/unix/thread.c

```diff
@@ -72,10 +75,59 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
 }
 
 
+int uv_thread_create_on(int proc_num,
+                        uv_thread_t *tid,
+                        void (*entry)(void *arg),
+                        void *arg) {
```

Comment:
> This can go one level below the versioning scheme, if you think that's still necessary.

The reason for versioning is that libuv can't change API or ABI in stable releases, only add them.

If `sizeof(uv_thread_affinity_t)` had to change for some reason, we'd have to bump the major version.  Downstream users like io.js would then have to bump their major version and io.js add-ons in turn would have to bump theirs as well.  It'd cause a lot of (avoidable) churn.

With that in mind, can you come up with an API that is versioned?

> Any comment on uv_thread_detach()?

Nope, looks fine to me.